docs/diagrams/wallet-lifecycle/wallet-lifecycle-3-responding-to-moving-funds-state.tikz

```diff
@@ -0,0 +1,153 @@
+% `convert -density 300 <filename>.pdf -resize 50% <filename>.png`.
+% This requires installation of the `imagemagick` tool:
+% `brew install imagemagick`.
+
+\documentclass{standalone}%
+\usepackage{tikz}%
+\usetikzlibrary{positioning,shapes,arrows.meta,chains}%
+
+% These commands are used to draw arrow labels
+\newcommand{\yes}{node [auto, pos=0, xshift=5mm]{Y}}%
+\newcommand{\no}{node [auto, swap, pos=0, xshift=-5mm]{N}}%
+
+% We need layers to draw the background
+\pgfdeclarelayer{background}%
+\pgfsetlayers{background,main}%
+
+\begin{document}%
+\begin{tikzpicture}[
+% Specify styles for nodes:
+    base/.style = {rectangle, rounded corners, draw=black,
+                   minimum width=4cm, minimum height=1cm,
+                   text centered, text width=4cm, align=flush center},
+   start/.style = {base, fill=blue!30},
+  revert/.style = {base, fill=red!30},
+  action/.style = {base, fill=black!5},
+   check/.style = {diamond, aspect=1.2, draw=black, fill=black!5,
+                   minimum width=1cm, align=center},
+   note1/.style = {rectangle, draw, dashed, fill=green!10,
+                   align=flush left, text width=5cm},
+   note2/.style = {rectangle, draw, dashed, fill=cyan!10,
+                   align=flush left, text width=5cm},
+              > = Triangle,                  % defines the arrow tip style when `->` is used
+           font = \sffamily,
+]
+
+% RESET MOVING FUNDS TIMEOUT
+% Nodes
+\node[start]                                                       (rmft_start)   {\textbf{Reset moving funds timeout}};
+\node[check, below=of rmft_start]                                  (rmft_check1)  {Is wallet\\ MOVING\_FUNDS?};
+\node[check, below right=of rmft_check1]                           (rmft_check2)  {Has the\\ moving funds\\ commitment been already\\ submitted for\\ the wallet?};
+\node[check, below left=of rmft_check2]                            (rmft_check3)  {Is \# of live wallets equal 0?};
+\node[revert, below right=of rmft_check2]                          (rmft_revert1) {Transaction reverted};
+\node[check, below right=of rmft_check3]                           (rmft_check4)  {Has at least\\ moving\_funds\_timeout\_reset\_delay\\ passed since moving\_funds\_timeout\\ was reset for the wallet?};
+\node[revert, below left=of rmft_check4, xshift=-3cm]                           (rmft_revert2) {Transaction reverted};
+\node[action, below right=of rmft_check4]                          (rmft_action1) {Reset the moving\_funds\_timeout};
+\node[action, below=of rmft_action1]                               (rmft_action2) {Notification sent};
+\node[note1, right=of rmft_start.south east, anchor=south west]    (rmft_note1)   {Can be successfully called for wallets:\\ 
+                                                                                   \textbf{- MOVING\_FUNDS}};
+\node[note2, above=of rmft_note1.north, anchor=south, yshift=-8mm] (rmft_note2)   {Sender incentivized by:\\ \textbf{Reimbursement, if maintainer}};
```

Comment:
Let's skip the details about maintainers. It's not the case for every function and we do not want to get this diagram too complicated with explaining all the details. Also, the wallet can get slashed if they do not move funds before the timeout so resetting the timeout helps them to avoid slashing:

```suggestion
\node[note2, above=of rmft_note1.north, anchor=south, yshift=-8mm] (rmft_note2)   {Sender incentivized by:\\ \textbf{Ether reimbursed, avoids getting slashed}};
```